Sources/SwiftCompilerPlugin/CompilerPlugin.swift

```diff
@@ -74,12 +74,14 @@ extension CompilerPlugin {
         return type
       }
     }
-    return nil
+
+    let pluginPath = CommandLine.arguments.first ?? Bundle.main.executablePath ?? ProcessInfo.processInfo.processName
+    throw CompilerPluginError(message: "macro implementation type '\(moduleName).\(typeName)' could not be found in executable plugin '\(pluginPath)'")
   }
 
   // @testable
   public func _resolveMacro(moduleName: String, typeName: String) -> Macro.Type? {
-    resolveMacro(moduleName: moduleName, typeName: typeName)
+    try? resolveMacro(moduleName: moduleName, typeName: typeName)
```

Comment:
Any reason why we don’t want to make `_resoleMacro` `throws` as well. Just because it’s not needed for test now doesn’t mean it won’t be needed in tests later.

Also: Why do we need `_resolveMacro` at all. Couldn’t `resolveMacro` just be `@_spi(Testing)`?

Reply:
I don't think it's worth it. If we want it later, we can do it later.

> Why do we need _resolveMacro at all

I took your advice https://github.com/apple/swift-syntax/pull/1359#discussion_r1118823628 If you recommend `@_spi(Testing)` on `resolveMacro`, I'd do it now and update the test.

Reply:
Hmm, interesting. I didn’t know that I suggested it. I suppose this was before we introduced `@_spi(Testing)`, which I hadn’t thought of before. I like `@_spi(Testing)` better now.